### DISCUSSION TITLE
Remove records older than `n` hours

### DISCUSSION THREAD

**Discussion by varna** [supabase/supabase#9037] [Questions] [answered]
Hi,

I'm trying to remove all table items that are older than 2 weeks automatically.

Running SQL Editor like this does delete all older items:
```
DELETE FROM failures WHERE failures.created_at < now() - '1 hour'::interval;
```
But how do I run this automatically?

- I tried supabase functions with RPC.
- I tried Netlify `schedule` 
- I tried `client.from('failures').delete().lt('created_at', (new Date(Date.now() - 1000 * 60 * 60)))` 

I couldn't make anything work except for the SQL Editor manual run. Maybe I have problems with RLS? idk. I'm kind of stuck. I thought I got problems with timezones, but after testing over the weekend it is still the same. All of these methods seems to run and return nothing `{ data: [], error: null }`. No errors, no nothing.

**Comment by GaryAustin1** [comment, marked as answer]
You can use cron to do this.  Towards the end of this some one does what you are looking for.  
https://github.com/supabase/supabase/discussions/5604

**Reply by varna** [reply]
Thanks! This one seems to work after enabling the cron extension.
```SELECT cron.schedule('Purge', '*/5 * * * *', $$ DELETE from public.failures WHERE now() - created_at > interval '1 hour' $$);```
